Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -76,44 +76,68 @@
      * point with the greatest weight).
      */
     protected final float[] weight;
+
     /**
      * Index values identifying the points in the sample. See
      * {@link com.amazon.randomcutforest.store.IPointStore}.
      */
     protected final int[] pointIndex;
+
     /**
      * Sequence indexes of the points in the sample.
      */
     protected final long[] sequenceIndex;
+
     /**
      * The number of points in the sample when full.
      */
     protected final int capacity;
+
     /**
      * The number of points currently in the sample.
      */
     protected int size;
+
     /**
      * The decay factor used for generating the weight of the point. For greater
      * values of lambda we become more biased in favor of recent points.
      */
-    private final double lambda;
+    private double lambda;
+
+    /**
+     * The last timestamp when lambda was changed
+     */
+    private long lastUpdateOflambda = 0;
+
+    /**
+     * most recent timestamp, used to determine lastUpdateOfLambda
+     */
+    private long largestSequenceIndexSeen = 0;
```

Comment:
I might simplify this further and just go with `maxSequenceIndex`.

Reply:
done